src/Sylius/Component/Promotion/Checker/CouponsEligibilityChecker.php

```diff
@@ -46,8 +46,6 @@ public function isEligible(PromotionSubjectInterface $subject, PromotionInterfac
         }
 
         if (!$this->isCouponEligible($promotion, $subject)) {
-            $this->eventDispatcher->dispatch(SyliusPromotionEvents::COUPON_NOT_ELIGIBLE, new GenericEvent($promotion));
```

Comment:
I feel this is unneeded, as it does not need to display a message when the coupon does not belong to one of the active promotions. Just needs to return false.